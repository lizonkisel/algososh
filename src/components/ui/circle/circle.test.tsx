import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Circle } from './circle';

import { ElementStates } from '../../../types/element-states';

describe('Тестируем компонент Button', () => {
    it('Circle рендерится без букв', () => {
        const tree = renderer.create(<Circle />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится с буквами', () => {
        const tree = renderer.create(<Circle letter="text"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится с текстом в head', () => {
        const tree = renderer.create(<Circle head="head"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится с react-element в head', () => {
        const tree = renderer.create(<Circle head={<Circle />}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится с текстом в tail', () => {
        const tree = renderer.create(<Circle tail="tail"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится с react-element в tail', () => {
        const tree = renderer.create(<Circle tail={<Circle />}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится с index', () => {
        const tree = renderer.create(<Circle index={0}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится с пропом isSmall === true', () => {
        const tree = renderer.create(<Circle isSmall={true}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится в состоянии default', () => {
        const tree = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится в состоянии changing', () => {
        const tree = renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Circle рендерится в состоянии modified', () => {
        const tree = renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
