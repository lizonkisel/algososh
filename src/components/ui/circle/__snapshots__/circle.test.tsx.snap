// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем компонент Button Circle рендерится без букв 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится с react-element в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится с react-element в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тестируем компонент Button Circle рендерится с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      text
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится с пропом isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится с текстом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Button Circle рендерится с текстом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;
